tracks/elixir/exercises/list-ops/mentoring.md

```diff
@@ -0,0 +1,136 @@
+### Reasonable solutions
+
+```elixir
+defmodule ListOps do
+  @spec count(list) :: non_neg_integer
+  def count(l) do
+    do_count(l, 0)
+  end
+
+  defp do_count([], total) do
+    total
+  end
+
+  defp do_count([_h|t], total) do
+    do_count(t, total + 1)
+  end
+
+  @spec reverse(list) :: list
+  def reverse(l) do
+    do_reverse(l, [])
+  end
+
+  defp do_reverse([h|[]], []) do
+    h
+  end
+
+  defp do_reverse([], acc) do
+    acc
+  end
+
+  defp do_reverse([h | t], acc) do
+    do_reverse(t, [h | acc])
+  end
+
+  @spec map(list, (any -> any)) :: list
+  def map(l, f) do
+    do_map(l, f, [])
+  end
+
+  defp do_map([], _f, acc) do
+    acc |> reverse
```

Comment:
Relatively minor, but I prefer to be consistent with using or not using pipes for a single, standalone function call. All other functions (except `reverse/1` are called without pipes, like so: `reverse(acc)`